tla/etcdraft.tla

```diff
@@ -0,0 +1,931 @@
+    \* @typeAlias: MSG = [ wrapped: Bool, mtype: Str, mterm: Int, msource: Int, mdest: Int, RVReq: RVREQT, RVResp: RVRESPT, AEReq: AEREQT, AEResp: AERESPT ];
+    \* @type: MSG -> Int;
+    messages
+VARIABLE 
+    pendingMessages
+messageVars == <<messages, pendingMessages>>
+
+----
+\* The following variables are all per server (functions with domain Server).
+
+\* The server's term number.
+VARIABLE 
+    \* @type: Int -> Int;
+    currentTerm
+\* The server's state (Follower, Candidate, or Leader).
+VARIABLE 
+    \* @type: Int -> Str;
+    state
+\* The candidate the server voted for in its current term, or
+\* Nil if it hasn't voted for any.
+VARIABLE 
+    \* @type: Int -> Int;
+    votedFor
+serverVars == <<currentTerm, state, votedFor>>
+
+\* A Sequence of log entries. The index into this sequence is the index of the
+\* log entry. Unfortunately, the Sequence module defines Head(s) as the entry
+\* with index 1, so be careful not to use that!
+VARIABLE 
+    \* @type: Int -> [ entries: LOGT, len: Int ];
+    log
+\* The index of the latest entry in the log the state machine may apply.
+VARIABLE 
+    \* @type: Int -> Int;
+    commitIndex
+logVars == <<log, commitIndex>>
+
+\* The following variables are used only on candidates:
+\* The set of servers from which the candidate has received a RequestVote
+\* response in its currentTerm.
+VARIABLE 
+    \* @type: Int -> Set(Int);
+    votesResponded
+\* The set of servers from which the candidate has received a vote in its
+\* currentTerm.
+VARIABLE 
+    \* @type: Int -> Set(Int);
+    votesGranted
+\* @type: Seq(Int -> Set(Int));
+candidateVars == <<votesResponded, votesGranted>>
+
+\* The following variables are used only on leaders:
+\* The latest entry that each follower has acknowledged is the same as the
+\* leader's. This is used to calculate commitIndex on the leader.
+VARIABLE 
+    \* @type: Int -> (Int -> Int);
+    matchIndex
+VARIABLE
+    pendingConfChangeIndex
+leaderVars == <<matchIndex, pendingConfChangeIndex>>
+
+\* @type: Int -> [jointConfig: Seq(Set(int)), learners: Set(int)]
+VARIABLE 
+    config
+VARIABLE 
+    reconfigCount
+
+configVars == <<config, reconfigCount>>
+
+VARIABLE 
+    durableState
+
+\* End of per server variables.
+----
+
+\* All variables; used for stuttering (asserting state hasn't changed).
+vars == <<messageVars, serverVars, candidateVars, leaderVars, logVars, configVars, durableState>>
+
+
+----
+\* Helpers
+
+\* The set of all quorums. This just calculates simple majorities, but the only
+\* important property is that every quorum overlaps with every other.
+Quorum(c) == {i \in SUBSET(c) : Cardinality(i) * 2 > Cardinality(c)}
+
+\* The set of all quorums. This just calculates simple majorities, but the only
+\* important property is that every quorum overlaps with every other.
+Quorums ==
+    [ s \in SUBSET Server |-> {i \in SUBSET(s) : Cardinality(i) * 2 > Cardinality(s)} ]
```

Comment:
Why don't we use `Quorum(s)` here instead of repeating full expression?

Reply:
It would look like:

```
Quorums ==
    [ s \in SUBSET Server |-> Quorum(s) ]
```

Reply:
You're correct. `Quorums` just pre-calculate all quorums. We can always use `Quroum(s)` instead. 

`Quorums` was used in another spec we were working on (2-node raft). I guess it was merged here accidently. But it won't hurt as you can see, we don't use it in this spec at all.

Reply:
Removes unused `Quorums` from spec.